Tasks/UseDotNetV2/globaljsonfetcher.ts

```diff
@@ -0,0 +1,69 @@
+import * as fileSystem from "fs";
+import * as tl from 'azure-pipelines-task-lib/task';
+import { DotNetCoreVersionFetcher } from "./versionfetcher";
+import { VersionInfo } from "./models";
+
+export class globalJsonFetcher {
+
+    private workingDirectory: string;
+    private versionFetcher: DotNetCoreVersionFetcher = new DotNetCoreVersionFetcher(true);
+    /**
+     * The global json fetcher provider functionality to extract the version information from all global json in the working directory.
+     * @param workingDirectory 
+     */
+    constructor(workingDirectory: string) {
+        this.workingDirectory = workingDirectory;
+    }
+
+    /**
+     * Get all version information from all global.json starting from the working directory.     
+     */
+    public async GetVersions (): Promise<VersionInfo[]>{
+        var versionInformation: VersionInfo[] = new Array<VersionInfo>();
+        var versionStrings = this.getVersionStrings();
+        for (let index = 0; index < versionStrings.length; index++) {
+            const version = versionStrings[index];            
+            var versionInfo = await this.versionFetcher.getVersionInfo(version, "sdk", false); 
+            versionInformation.push(versionInfo);
+        }
+        return versionInformation
+            .sort((a,b) => a.getVersion().localeCompare(b.getVersion()));        
```

Comment:
Why is sorting required?

Reply:
this is requiered to install the sdk's in the right order to the `dotnet.exe` that is used from the path is the newest.

Reply:
Does the order matter? I believe with our current installer logic, the order of installation should not matter. We keep the dotnet.exe from the latest version. @hiyadav  to confirm

Reply:
yes I have find the magic line wich do that: 

```typescript
 if (this.packageType == utils.Constants.sdk && this.isLatestInstalledVersion(version))
```

It's in the `versioninstaller.ts` file in line 62.

I will remove my sorting.

Reply:
done